Updated Release and Main Pipeline (#3623)
ElektraInitiative/libelektra: doc/INSTALL.md

For [OpenSUSE, CentOS, Fedora, RHEL and SLE](https://build.opensuse.org/package/show/home:bekun:devel/elektra)
Kai-Uwe Behrmann kindly provides packages [for download](http://software.opensuse.org/download.html?project=home%3Abekun%3Adevel&package=libelektra4).

> We also provide more recent versions and bleeding-edge versions in our repositories:

Reviewer: "more recent" and "bleeding-edge" unclear. Isn't it latest release and latest build of master?

### Debian/Ubuntu

> We provide repositories for latest releases (stable) and latest builds from master (unstable) for following Debian-based distributions:

Reviewer: ```suggestion
We provide repositories for latest releases and latest builds from master (suite postfixed with `-unstable`) for following Debian-based distributions:
```

- Debian Buster
- Ubuntu Focal
- Ubuntu Bionic

To use our stable repositories with our latest releases, following steps need to be made:

1. Run `sudo apt-key adv --keyserver keys.gnupg.net --recv-keys F26BBE02F3C315A19BF1F791A9A25CC1CC83E839` to obtain the key.

> 2. Add `deb https://debs.libelektra.org/<DISTRIBUTION> <DISTRIBUTION> main` into `/etc/apt/sources.list`

Reviewer: ```suggestion
2. Add `deb https://debs.libelektra.org/<DISTRIBUTION> <SUITE> main` into `/etc/apt/sources.list`
```

>    where `<DISTRIBUTION>` is the Codename of your distributions e.g.`focal`,`bionic`,`buster`, etc.

Reviewer: ```suggestion
   where `<DISTRIBUTION>` and `<SUITE>` is the Codename of your distributions e.g.`focal`,`bionic`,`buster`, etc.
```

This can also be done using:

```sh
# Example for Ubuntu Focal
apt-get install software-properties-common apt-transport-https
echo "deb https://debs.libelektra.org/focal focal main" | sudo tee /etc/apt/sources.list.d/elektra.list
```

Or alternatively, you can use (if you do not mind many dependences just to add one line to a config file):

```sh
# Example for Ubuntu Focal
sudo apt-get install software-properties-common apt-transport-https
sudo add-apt-repository "deb https://debs.libelektra.org/focal focal main"
```

> If you would like to use the latest builds of master, append `-unstable` to `<DISTRIBUTION>`.

Reviewer: ```suggestion
If you would like to use the latest builds of master, append `-unstable` to `<SUITE>`.
```

The `etc/apt/source.list` entry must look like following: `deb https://debs.libelektra.org/<DISTRIBUTION> <DISTRIBUTION>-unstable main`

E.g. `deb https://debs.libelektra.org/focal focal-unstable main`

> > NOTE: for Ubuntu Bionic the yamlcpp plugin is excluded due to missing dependencies and therefore the package `libelektra4-yamlcpp` is not available.

Reviewer: Is this really the only plugin (or tool or binding) which is excluded? Did you check in the log if everything else is included?

Author: I checked the logs and all plugins/bindings/tools that are part of a package are included on all distros except for the yamlccp plugin on Ubuntu Bionic.

Reviewer: Thank you so much for checking this! We should also add a TODO task for the releases to check this for every release.

### Fedora

> We also provide repositories for latest releases (stable) and latest builds from master (unstable) for Fedora 33.

Reviewer: Maybe here similar fixes as above.

For the stable repository:

```sh
wget https://rpms.libelektra.org/fedora-33/libelektra.repo -O libelektra.repo;
sudo mv libelektra.repo /etc/yum.repos.d/;
sudo yum update
```

Or alternatively you can use dnf to add this repo:

```sh
dnf config-manager --add-repo https://rpms.libelektra.org/fedora-33/libelektra.repo
```

For our bleeding-edge builds append `-unstable` to the distribution name:

```sh
wget https://rpms.libelektra.org/fedora-33-unstable/libelektra.repo -O libelektra.repo;
sudo mv libelektra.repo /etc/yum.repos.d/;
sudo yum update
```

Or alternatively you can use dnf to add this repo:

```sh
dnf config-manager --add-repo https://rpms.libelektra.org/fedora-33-unstable/libelektra.repo
```

### Install

To get all packaged plugins, bindings and tools install:

```sh
# For Debian based distributions

> apt-get install libelektra4-all

Reviewer: We should pump this major version number (and SO_VERSION) to 5 to be ready for 1.0 release. This is also a good indicator for external people to indicate that this will be a "special" release with huge changes as @kodebach suggested.

Author: Do you mean with major version number the "4" in libelektra4-all, etc? If yes, should I change the SO_VERSION including the package version numbers already in this PR?

Reviewer: Exactly, I mean this major shared object (SO) number for the core library which is both in the package names (like libelektra**4**-all) and in the shared objects (as defined by SO_VERSION).

It is up to you in which PR you want to do it.

Author: I suppose the installation of of a libelektra5-* package should break/replace libelektra4-* packages? Or should it be possible to install both? (Unfortunately I'm not that familiar with SO versions.)

Reviewer: I am not an expert in Debian packaging but I think you are right, e.g.:

```
apt-cache show libpurelibc1                                                                                                                                                                             Package: libpurelibc1
Source: purelibc
Version: 0.4.1-2
Installed-Size: 74
Maintainer: Debian VSquare Team <virtualsquare@cs.unibo.it>
Architecture: amd64
Replaces: libpurelibc0
Depends: libc6 (>= 2.14)
Conflicts: libpurelibc0
```

So a "Replaces:" and "Conflicts:" seems to be the right thing.

> Or should it be possible to install both?

This sounds like a lot of effort, I do not think this is needed.

# For Fedora based distributions
dnf install libelektra4-all
```

For a small installation with command-line tools available use:

```sh
# For Debian based distributions
apt-get install elektra-bin
# For Fedora based distributions
dnf install elektra-bin
```

> To install all debugsym/debuginfo packages:

Reviewer: Is this included in `libelektra4-all`?

Author: Yes but only as a "Suggests" dependency and not as a "Depends". On RPM this is included as a "Recommends" dependency.

Reviewer: Ok, then the text is okay as it is.
